java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -426,12 +435,28 @@ private Resolution findMethod(Env env, JavaType callSite, JavaType site, String
     return bestSoFar;
   }
 
+  private Resolution lookupInScope(Env env, JavaType callSite, JavaType site, String name, List<JavaType> argTypes, List<JavaType> typeParams,
+                                   boolean autoboxing, Scope scope, Resolution bestFound) {
+    Resolution bestSoFar = bestFound;
+    // look in site members
+    for (JavaSymbol symbol : scope.lookup(name)) {
+      if (symbol.kind == JavaSymbol.MTH) {
+        Resolution best = selectBest(env, site, callSite, argTypes, typeParams, symbol, bestSoFar, autoboxing);
+        if (best.symbol == symbol) {
+          bestSoFar = best;
+        }
+      }
+    }
+    return bestSoFar;
+  }
+
   /**
    * @param candidate    candidate
    * @param bestSoFar previously found best match
    */
-  private Resolution selectBest(Env env, JavaType site, List<JavaType> argTypes, List<JavaType> typeParams, JavaSymbol candidate, Resolution bestSoFar, boolean autoboxing) {
-    JavaSymbol.TypeJavaSymbol siteSymbol = site.symbol;
+  private Resolution selectBest(Env env, JavaType defSite, JavaType callSite, List<JavaType> argTypes, List<JavaType> typeParams,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)